internal/config/datasources.go

```diff
@@ -0,0 +1,39 @@
+package config
+
+import (
+	"github.com/spf13/viper"
+
+	"github.com/anchore/grype/grype/matcher/java"
+)
+
+const (
+	defaultMavenBaseURL = "https://search.maven.org/solrsearch/select"
+)
+
+type externalSources struct {
+	Enable bool  `yaml:"enable" json:"enable" mapstructure:"enable"`
+	Maven  maven `yaml:"maven" json:"maven" mapsructure:"maven"`
+}
+
+type maven struct {
+	SearchMavenUpstream bool   `yaml:"search-maven-upstream" json:"search_maven_upstream" mapstructure:"search-maven-upstream"`
```

Comment:
nit: the full config path stutters: `external-sources.maven.search-maven-upstream`
maybe rename the field to `search-upstream`? Or more descriptive to the what or how of the operation, such as `search-by-sha1` or `find-artifact-group-id` (not saying these suggestions are necessarily better, but wanted to give an idea)

Reply:
Yea nice suggestion. I'll update the config here.